lib/koans/18_genservers.ex

```diff
@@ -0,0 +1,68 @@
+defmodule GenServers do
+  use Koans
+
+  @intro "GenServers"
+
+  defmodule ExampleGenServer do
+    use GenServer
+
+    # Use init to set up initial state []
+    def init(_args) do
+      {:ok, []}
+    end
+
+    # Use calls to perform synchronous actions
+    def handle_call(:pop, _from, [head|tail]) do
+      {:reply, head, tail}
+    end
+
+    def handle_call(:peek, _from, stack) do
+      {:reply, stack, stack}
+    end
+
+    # Use casts to perform asynchronous actions
+    def handle_cast({:push, value}, stack) do
+      {:noreply, [value|stack]}
+    end
+  end
+
+
+  koan "You can think of GenServers like more flexible agents" do
+    {:ok, pid} = GenServer.start_link(ExampleGenServer, :ok, [])
+
+    assert is_pid(pid) == ___
+  end
+
+  koan "GenServers can have a name too" do
+    {:ok, pid} = GenServer.start_link(ExampleGenServer, :ok, name: StackServer)
+
+    assert is_pid(GenServer.whereis(StackServer)) == ___
```

Comment:
👍  for that as well. I will add/refactor some koans to make handling more pid related, as @Nagasaki45 mentioned as well.